lib/gateways/airtable-gateway.js

```diff
@@ -10,27 +10,38 @@ class AirtableGateway {
   }
 
   async all() {
-    const results = await Promise.all(
-      this.tableNames.map(table => {
-        return this.tables(table).select().all();
-      })
-    );
-
-    let categorisedResources = [];
-
-    results.flat().forEach(result =>
-      result.fields.Categories?.forEach(category => {
-        categorisedResources.push({ ...result, categoryName: category, tag: "Council" });
-      })
-    );
-
-    return categorisedResources.map(createResourceFromModel);
+    try{
+      const results = await Promise.all(
+        this.tableNames.map(table => {
+          return this.tables(table).select().all();
+        })
+      );
+  
+      let categorisedResources = [];
+  
+      results.flat().forEach(result =>
+        result.fields.Categories?.forEach(category => {
+          categorisedResources.push({ ...result, categoryName: category, tag: "Council" });
+        })
+      );
+  
+      return categorisedResources.map(createResourceFromModel);
+    } catch (err){
+      console.log("Error fetching resources", err);
+      return [];
```

Comment:
Hey @kosiakkatrina 

I'd recommend being able to distinguish between "no resources" and "an error occurred" so the application or frontend can show an appropriate message to the user.

From the gateway, consider returning an object that indicates success or failure, along with the payload.

```javascript
// Error scenario
return {
  data: [],
  error: "Error fetching resources"
};
// Success scenario
return {
  data: theResources,
  error: null
};
```

This can then be used further up the call stack to show a warning.

```
const result = resourcesGateway.fetch(...);

if (result.error) {
  // Handle error, set a prop, etc
}
```